### PR TITLE
KINNO-480-app_item_layout

### PR DESCRIPTION
**Pull request by shayb**
add first list apps list with loader and no data display

### CODE REVIEW

File: appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/presenter/AppsDiscoveryListPresenter.kt [hunk 1]
```diff
@@ -8,6 +8,18 @@ import org.kinecosystem.appsdiscovery.sender.repositories.OperationResultCallbac
...
+    private var loadingListener:LoadingListener? = null
+
+    interface LoadingListener{
+        fun loading()
+        fun loadingComplete()
+        fun loadingFailed()
+    }
+
+    override fun setLoadingListener(loadingListener: LoadingListener) {
+        this.loadingListener = loadingListener
```
Comment (berryve) [inline appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/presenter/AppsDiscoveryListPresenter.kt:20]: you need to make sure the listener is set to null when the activity is destroyed

File: appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/view/AppsDiscoveryActivity.kt [hunk 2]
```diff
@@ -3,14 +3,61 @@ package org.kinecosystem.appsdiscovery.sender.discovery.view
...
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.apps_discovery_activity)
+        dataGroup = findViewById(R.id.data)
+        noDataGroup = findViewById(R.id.noData)
+        loader = findViewById(R.id.loader)
+        findViewById<AppsDiscoveryList>(R.id.list).setLoadingListener(object : AppsDiscoveryListPresenter.LoadingListener {
+            override fun loading() {
+                updateLoading()
```
Comment (berryve) [inline appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/view/AppsDiscoveryActivity.kt:28]: here by calling updateLoading here you actually add the activity and it's views as a member variable of your listener.  You pass this listener to a presenter and model that might outlive the activity and thus continue holding a reference to it after it has already been destroyed. Therefore you need to make sure you remove the listener when activity is destroyed.

File: appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/presenter/AppsDiscoveryListPresenter.kt (hunk 1 above)
Comment (shayb) [reply appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/presenter/AppsDiscoveryListPresenter.kt]: 👍

File: appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/view/AppsDiscoveryActivity.kt (hunk 2 above)
Comment (shayb) [reply appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/view/AppsDiscoveryActivity.kt]: 👍

Comment (shayb) [reply appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/view/AppsDiscoveryActivity.kt:28]: berry if the activity is destroyed then the list is detached?
if hold ref to the activity but its a view that has a ref to the parent activity
what do you think?

File: appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/presenter/AppsDiscoveryListPresenter.kt (hunk 1 above)
Comment (shayb) [reply appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/presenter/AppsDiscoveryListPresenter.kt:20]: on detach i  reset it to null

File: appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/view/AppsDiscoveryActivity.kt (hunk 2 above)
Comment (shayb) [reply appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/view/AppsDiscoveryActivity.kt:28]: i reset the listener to null when detached

**Pull request merged**